Tools/ardupilotwaf/chibios.py

```diff
@@ -65,10 +65,16 @@ def keyword(self):
 
 class set_default_parameters(Task.Task):
     color='CYAN'
-    run_str='python ${APJ_TOOL} --set-file ${DEFAULT_PARAMETERS} ${SRC}'
     always_run = True
     def keyword(self):
         return "apj_tool"
+    def run(self):
+        rel_default_parameters = self.env.get_flat('DEFAULT_PARAMETERS')
+        abs_default_parameters = os.path.join(self.env.STANDARD_BUILDDIR,
```

Comment:
I'm confused about this:

- DEFAULT_PARAMETERS should be an absolute path
- looks to me like STANDARD_BUILDDIR won't exist (how does *some-board* get converted to the actual board?), especially if the build directory isn't the standard one.

Reply:
Yes, this is all because `DEFAULT_PARAMETERS` is NOT  an absolute path.  It's an "environment variable" defined as part of the hwdef.dat

STANDARD_BUILDDIR will definitely not exist.  But it's somewhere that someonewho isn't passing in `--out` would typically build, and this `DEFAULT_PARAMETERS`  will be relative to.

Reply:
I'm still confused. DEFAULT_PARAMETERS is set at https://github.com/ArduPilot/ardupilot/blob/6652dd73e4c97f83325a85f3a8e321abf6eed2f2/Tools/ardupilotwaf/chibios.py#L207
which uses srcpath
https://github.com/ArduPilot/ardupilot/blob/6652dd73e4c97f83325a85f3a8e321abf6eed2f2/Tools/ardupilotwaf/chibios.py#L171-L172
which gives an absolute path.

I get what you are trying to do with STANDARD_BUILDDIR though: since DEFAULT_PARAMETERS would be relative, it doesn't matter if STANDARD_BUILDDIR exists.

Reply:
OK, I understand the issue now, it's loading the DEFAULT_PARAMETERS from the env.py file, which comes from hwdef.dat file and when it does this it doesn't go through the srcpath method.

I'm not sure what is the perfect way to fix this, but I don't like this way. I'll think about it a bit more.